### PR TITLE
added binary_count_trailing_zeros.py

### PR DESCRIPTION
**Pull request by vivek9patel**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: bit_manipulation/binary_count_trailing_zeros.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    Traceback (most recent call last):
+        ...
+    TypeError: unsupported operand type(s) for &: 'float' and 'float'
+    >>> binary_count_trailing_zeros("0")
+    Traceback (most recent call last):
+        ...
+    TypeError: '<' not supported between instances of 'str' and 'int'
+    """
+    if a < 0:
+        raise ValueError("the value of input must be positive")
```
Comment (dhruvmanila) [inline bit_manipulation/binary_count_trailing_zeros.py:35]:
```suggestion
    if a < 0:
        raise ValueError("Input value must be a positive integer")
```
Please use `isinstance` to check whether the input is of type `float` and raise `TypeError` with an appropriate message.

File: bit_manipulation/binary_count_trailing_zeros.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+from math import log2
+
+
+def binary_count_trailing_zeros(a: int):
```
Comment (dhruvmanila) [inline bit_manipulation/binary_count_trailing_zeros.py:4]:
```suggestion
def binary_count_trailing_zeros(a: int) -> int:
```

File: bit_manipulation/binary_count_trailing_zeros.py (hunk 1 above)
Comment (vivek9patel) [reply bit_manipulation/binary_count_trailing_zeros.py:35]: sure!

File: bit_manipulation/binary_count_trailing_zeros.py (hunk 2 above)
Comment (vivek9patel) [reply bit_manipulation/binary_count_trailing_zeros.py]: @dhruvmanila I have done the required changes!

**Comment by vivek9patel** [comment]
@cclauss can you please review this?

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

**Pull request merged**